Sources/Swift/Integrations/WatchdogTerminations/Processors/SentryWatchdogTerminationContextProcessor.swift

```diff
@@ -0,0 +1,42 @@
+@_implementationOnly import _SentryPrivate
+import Foundation
+
+@objcMembers
+class SentryWatchdogTerminationContextProcessor: NSObject {
+
+    private let dispatchQueueWrapper: SentryDispatchQueueWrapper
+    private let scopeContextStore: SentryScopeContextPersistentStore
+
+    init(
+        withDispatchQueueWrapper dispatchQueueWrapper: SentryDispatchQueueWrapper,
+        scopeContextStore: SentryScopeContextPersistentStore
+    ) {
+        self.dispatchQueueWrapper = dispatchQueueWrapper
+        self.scopeContextStore = scopeContextStore
+
+        super.init()
+
+        clear()
```

Comment:
`m`: I don't think we need to call `clear` in init actually, because we anyways overwrite the context here https://github.com/getsentry/sentry-cocoa/blob/76fc6d930d0da34f652ff03eeae6e1d14ca3f072/Sources/Sentry/SentryWatchdogTerminationTrackingIntegration.m#L103-L105

This an extra file IO operation on the calling thread, and I'd like to avoid it.

If we want to call clear I would rather call it in `SentryWatchdogTerminationTrackingIntegration`, cause doing that in init seems a bit strange to me.

Reply:
Ah I think I get why we need it, `setContext` happens async, so we could end up in a dirty state.

Reply:
The `SentryWatchdogTerminationBreadcrumbProcessor` was my reference here. It is calling `switchFileHandle` on `init` which also removes and creates a new file for the active path. This is the same logic but written differently.